Sources/Base/Types.swift

```diff
@@ -166,20 +192,20 @@ func ==(lhs: DataType, rhs: DataType) -> Bool {
 }
 
 class Decl: ASTNode {
-  var type: DataType
+  var type: DataType?
```

Comment:
@harlanhaskins Either way I expect queries before typechecking to blow up and queries after to not have to see error types in the AST.  It might even be worth it to create a second fully-types AST that is suitable for lowering.